migrator-tool/src/main/scala/za/co/absa/spline/migrator/Spline03Actor.scala

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.migrator
+
+import java.util
+import java.util.UUID
+
+import akka.actor.{Actor, ActorLogging, ActorRef}
+import akka.pattern.pipe
+import com.mongodb.ConnectionString
+import com.mongodb.casbah.Imports.DBObject
+import com.mongodb.client.model.Aggregates.{`match`, addFields, project}
+import com.mongodb.client.model.Field
+import com.mongodb.client.model.Filters.and
+import com.mongodb.client.model.Projections.{fields, include}
+import org.bson.Document.parse
+import za.co.absa.spline.common.ARM.managed
+import za.co.absa.spline.migrator.Spline03Actor._
+import za.co.absa.spline.model.DataLineageId.toDatasetId
+import za.co.absa.spline.model.{DataLineage, PersistedDatasetDescriptor}
+import za.co.absa.spline.persistence.api.CloseableIterable
+import za.co.absa.spline.persistence.api.DataLineageReader.PageRequest
+import za.co.absa.spline.persistence.mongo.dao.{LineageDAOv3, LineageDAOv4, MultiVersionLineageDAO}
+import za.co.absa.spline.persistence.mongo.{MongoConnectionImpl, MongoDataLineageReader}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+
+
+object Spline03Actor {
+
+  trait RequestMessage
+
+  case class GetExistingLineages(page: PageRequest) extends RequestMessage
+
+  case object GetFutureLineages extends RequestMessage
+
+
+  trait ResponseMessage
+
+  case class PageSize(count: Int) extends ResponseMessage
+
+  case class DataLineageLoaded(lineage: DataLineage) extends ResponseMessage
+
+  case class DataLineageLoadFailure(lineageId: UUID, e: Throwable) extends ResponseMessage
+
+}
+
+class Spline03Actor(connectionUrl: String) extends Actor with ActorLogging {
+  implicit val ec: ExecutionContext = context.dispatcher
+
+  private val mongoConnection =
+    new MongoConnectionImpl(connectionUrl)
+
+  private val mongoDatabase = {
+    val dbName = new ConnectionString(connectionUrl).getDatabase
+    mongoConnection
+      .db
+      .underlying
+      .getMongoClient
+      .getDatabase(dbName)
+  }
+
+  private val mongoReader =
+    new MongoDataLineageReader(
+      new MultiVersionLineageDAO(
+        new LineageDAOv3(mongoConnection),
+        new LineageDAOv4(mongoConnection)))
+
+
+  override def receive: Receive = {
+    case GetExistingLineages(page) =>
+      val theSender = sender
```

Comment:
Because `sender` is a method and `managed` takes a closure, effectively taking the `sender` invocation out of the message handling context and potentially the current thread. This may lead to an inconsistent non-deterministic behaviour and is forbidden by Akka's docs.

> * WARNING: Only valid within the Actor itself, so do not close over it and
> * publish it to other threads!